rstudio/rstudio
# make Shift+Command+Zero the sole shortcut for restartR on Mac

### Intent

Prompted by #7695

On Mac there are two shortcuts for Restart R. <kbd>Shift+Command+F10</kbd> (which is the shortcut on Windows and Linux as well), and the undocumented Mac-only <kbd>Shift+Command+0</kbd>.

<kbd>Shift+Command+F10</kbd> is shown in the keyboard shortcuts dialog, on the Session menu, and in the keyboard shortcut help, though it is shown as <kbd>Shift+Command+0</kbd> on the command palette. This is makes it unclear what to do if a user wants to change or remove this shortcut.

Initially proposed removing <kbd>Shift+Command+0</kbd> shortcut altogether, but in a poll of users, those who knew about it strongly preferred it, and used it frequently.

### Approach

On Mac make <kbd>Shift+Command+0</kbd> the only shortcut for `restartR` command, removing <kbd>Shift+Command+F10</kbd>.

### Automated Tests

No existing tests for individual keyboard shortcuts. I checked and we don't reference F10 in any of our Selenium automation so this shouldn't break, even if run on Mac.

### QA Notes

Verify that hitting <kbd>Cmd+Shift+0</kbd> on Mac triggers Restart R, and is shown in shortcuts dialog, Session menu, command palette, and keyboard shortcut help (both variations, the popup shown when screen reader accessibility is off, the other, a web page, shown when screen reader is enabled).

### Checklist
